### PR TITLE
DOC: Add plotting examples and fix broken examples

### PR DESCRIPTION
**Pull request by deppen8**
This PR improves the `pandas.plotting._misc.py` module docs by:
1. Adding example plots where they were missing from docstrings.
2. Fixing examples that were not formatted to render with
```
.. plot::
    :context: close-figs
```
3. Simplifying the syntax of some oddly-written examples.

In some cases, I replicated the examples found in the [User Guide](https://pandas.pydata.org/pandas-docs/stable/user_guide/visualization.html#plotting-tools).

### CODE REVIEW

**Comment by datapythonista** [comment]
Thanks @deppen8, cool stuff. Did you render the website to see that everything is looking as expected?

@jorisvandenbossche, may be you want to have a look here.

File: pandas/plotting/_misc.py [hunk 1]
```diff
@@ -298,10 +322,11 @@ def bootstrap_plot(series, fig=None, size=50, samples=500, **kwds):
     This example draws a basic bootstap plot for a Series.
 
     .. plot::
-            :context: close-figs
+        :context: close-figs
 
-            >>> s = pd.Series(np.random.uniform(size=100))
-            >>> fig = pd.plotting.bootstrap_plot(s)  # doctest: +SKIP
+        >>> np.random.seed(5)
```
Comment (jorisvandenbossche) [inline pandas/plotting/_misc.py:327]: is this seed needed?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! Looks good generally, just wondering: are the random seeds necessary? We don't really do this in other parts of docs, I think.

**Comment by deppen8** [comment]
@datapythonista, I built the docs locally and nothing looked out of order.

@jorisvandenbossche I found this advice which suggested using random seeds.

https://pandas.pydata.org/docs/development/contributing_docstring.html#tips-for-getting-your-examples-pass-the-doctests

In the case of the lag plot, it has an added benefit of keeping the demo data above zero, which probably matches most use cases better.

**Comment by jorisvandenbossche** [comment]
> I found this advice which suggested using random seeds.

Ah, yes, but that's only to get doctests passing where eg a number is shown in the output, while here the plots are not actually checked.

For the lag plot that sounds like a good reason, for the others I would maybe remove it again, if we don't do that elsewhere

**Comment by deppen8** [comment]
> For the lag plot that sounds like a good reason, for the others I would maybe remove it again, if we don't do that elsewhere

Done!

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @deppen8 !

**Pull request merged**